backend/src/db/repositories/CommentRepository.ts

```diff
@@ -68,10 +72,13 @@ export default class CommentRepository {
         });
     }
 
-    async getUserCommentsTotal(userId: number): Promise<number> {
-        return this.db.fetchOne<{ cnt: string }>('select count(*) cnt from comments where author_id = :user_id and deleted = 0', {
-            user_id: userId
-        }).then((res) => parseInt(res.cnt || '0'));
+    async getUserCommentsTotal(userId: number, filter: string): Promise<number> {
+        const query = `select count(*) cnt from comments where author_id = ? and deleted = 0 ${filter ? '  and source like ? ' : ''}`;
+        const params: Array<number | string> = [userId];
+        if (filter) {
```

Comment:
Regarding the conditional params. I just tested, and it seems that you can have unused params with `undefined` value:

```
{
              username: username,
              'test': undefined
          }
```
Should reduce the boilerplate.

Reply:
Fixed.